### PR TITLE
Move rproc ops to separate file

### PR DESCRIPTION
**Pull request by bentheredonethat**
previously, only zynq has separate file for rproc operations. 

In zynqmp and r5 platform, move rproc operations to separate file from platform info to separate file to make the format consistent.

### CODE REVIEW

File: apps/machine/zynqmp_r5/platform_info.c [hunk 1]
```diff
@@ -97,149 +97,9 @@ static struct remoteproc rproc_inst;
 extern int init_system(void);
 extern void cleanup_system(void);
 
-static int zynqmp_r5_a53_proc_irq_handler(int vect_id, void *data)
```
Comment (wjliang) [inline apps/machine/zynqmp_r5/platform_info.c]: IPI reg macros should also been moved to the remoteproc implementation file, struct remoteproc_priv() should definition should either move to remoteproc driver or define in the platform info header file

Comment (bentheredonethat) [reply apps/machine/zynqmp_r5/platform_info.c]: done with rebased commits.

File: apps/machine/zynqmp_r5/platform_info.h [hunk 2]
```diff
@@ -9,9 +9,26 @@
...
 #endif
 
+/* IPI REGs OFFSET */
+#define IPI_TRIG_OFFSET          0x00000000    /* IPI trigger register offset */
+#define IPI_OBS_OFFSET           0x00000004    /* IPI observation register offset */
+#define IPI_ISR_OFFSET           0x00000010    /* IPI interrupt status register offset */
+#define IPI_IMR_OFFSET           0x00000014    /* IPI interrupt mask register offset */
+#define IPI_IER_OFFSET           0x00000018    /* IPI interrupt enable register offset */
+#define IPI_IDR_OFFSET           0x0000001C    /* IPI interrupt disable register offset */
+
```
Comment (wjliang) [inline apps/machine/zynqmp_r5/platform_info.h]: This macros should be defined in the remoteproc file as it doesn't need  to be shared

File: apps/machine/zynqmp_r5/zynqmp_r5_a53_rproc.c [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+#define IPI_IER_OFFSET           0x00000018    /* IPI interrupt enable register offset */
+#define IPI_IDR_OFFSET           0x0000001C    /* IPI interrupt disable register offset */
+
+/* Cortex R5 memory attributes */
+#define DEVICE_SHARED       0x00000001U /* device, shareable */
+#define DEVICE_NONSHARED    0x00000010U /* device, non shareable */
+#define NORM_NSHARED_NCACHE 0x00000008U /* Non cacheable  non shareable */
+#define NORM_SHARED_NCACHE  0x0000000CU /* Non cacheable shareable */
+#define PRIV_RW_USER_RW     (0x00000003U<<8U) /* Full Access */
+
```
Comment (wjliang) [inline apps/machine/zynqmp_r5/zynqmp_r5_a53_rproc.c]: These macros are also used in platform_info.c, please move them to platform_info.h

Comment (bentheredonethat) [reply apps/machine/zynqmp_r5/zynqmp_r5_a53_rproc.c]: updated in rebased commits.

**Pull request merged**